ksqldb-parser/src/main/java/io/confluent/ksql/util/QueryMask.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ * http://www.confluent.io/confluent-community-license
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.util;
+
+import com.google.common.collect.ImmutableSet;
+import io.confluent.ksql.parser.DefaultKsqlParser;
+import io.confluent.ksql.parser.SqlBaseBaseVisitor;
+import io.confluent.ksql.parser.SqlBaseParser.CreateConnectorContext;
+import io.confluent.ksql.parser.SqlBaseParser.SingleStatementContext;
+import io.confluent.ksql.parser.SqlBaseParser.StatementsContext;
+import io.confluent.ksql.parser.SqlBaseParser.TablePropertiesContext;
+import io.confluent.ksql.parser.SqlBaseParser.TablePropertyContext;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.antlr.v4.runtime.misc.Interval;
+import org.antlr.v4.runtime.tree.ParseTree;
+import org.apache.commons.lang3.StringUtils;
+
+public final class QueryMask {
+
+  private static final ImmutableSet<String> ALLOWED_KEYS = ImmutableSet.of("connector.class");
+  private static final String MASKED_VALUE = "'[string]'";
+
+  private QueryMask() {}
+
+  public static String getMaskedStatement(final String query) {
+    try {
+      final ParseTree tree = DefaultKsqlParser.getParseTree(query);
+      return new Visitor().visit(tree);
+    } catch (final Exception e) {
+      return maskConnectProperties(query);
+    }
+  }
+
+  private static String maskConnectProperties(final String query) {
+    /*
+      Failed to parse statement. Mask disallowed keys by string replacing. This is a best effort
+      attempt. The assumption is that properties are in right format: it matches key wrapped in
+      single quotes equals value wrapped in single quotes. Key and value could be empty and there
+      could be spaces around equal sign.
+     */
+    final StringBuffer sb = new StringBuffer();
+    final Pattern pattern = Pattern.compile("('[^']+'|\"[^\"]+\")\\s*=\\s*('[^']*')",
```

Comment:
Huh interesting... up until now I thought it was invalid syntax for CREATE CONNECTOR table properties to double-quote the keys (I thought only single-quotes were allowed) but upon closer inspection it looks like we allow not only double-quotes but also backticks. Does that match what you're seeing as well? If so, let's add handling for backticks into this regex too.